### DISCUSSION TITLE
Enter key maps to j

### DISCUSSION THREAD

**Discussion by migsar** [helix-editor/helix#8982] [Troubleshooting]
### Summary

I tried my best to figure out this on my own, maybe it is mentioned somewhere but I didn't think of the right keywords.

When I enter the editor, in Normal mode, `enter` key maps to `j`, in Insert mode it works fine. I have to use `shift+enter` to execute any command.

I discovered it may be related to Alacritty, since in Tilix it works fine. Nothing is logged, so I am not sure if it is a Helix thing or an Alacritty thing.

### Reproduction Steps

 

I tried this:

1. `hx`
2. Pressed `:q` to quit the editor.
3. When I press `enter`, `:qj` appears in the command bar, I have to press `shift+enter` to execute the command.

I expected this to happen:

Close the application.

Instead, this happened:

A `j` character appears after the command I'm typing as part of the command.

### Helix log

No problem being logged.

### Platform

Linux Ubuntu 22.04.3 LTS

### Terminal Emulator

alacritty 0.10.1 ()

### Installation Method

apt package manager

### Helix Version

helix 23.10 (f6021dd0)

**Comment by pascalkuthe** [comment]
I can't reproduce this, I also use alacritty and never had any similar issues on a wide variety of distros and helic versions. Perhaphs something is wrong with your setup or alacritty installation. Nothing we can do on our end if it's not reproducible

**Reply by migsar** [reply]
Thanks for the quick response, it makes sense. Any idea of additional steps or information I can provide to make it reproducible?

**Comment by srprca** [comment]
Hello!

I am hitting exactly this problem when I run Helix `25.07.1` (`a05c151b`) inside a `fedora` (`ba3f1ab896c1`) or `fedora-minimal` (`22f1cfa48f4b`) Podman container from `registry.fedoraproject.org` under Fedora 43 host, both in Alacritty `0.16.1` and in Konsole `25.12.2`.

In my case, even `Shift+Enter` doesn't help. I tried installing `ncurses-term` for Terminfo definitions, but this didn't help.

Please let me know if you want me to submit this as a bug instead.

Thank you!

**Reply by srprca** [reply]
The same issue occurs with a Fedora Silverblue container (`quay.io/fedora/fedora-silverblue`, `bec549b25e4e`), which presumably has all packages for proper functioning of terminal applications installed (I am not sure whether this holds for `fedora` and `fedora-minimal`, as they are pretty slim). So I guess this might be something about environment variables or terminal settings...?

**Reply by srprca** [reply]
I tried `export`ing some environment variables inside the container to have the same values as they do on the host system, and also adding one `stty` flag that was absent inside the container:
```
# Later changed to C.UTF-8 because the locale was not present
export LANG=en_US.UTF-8
export TERM=xterm-256color
export LANG=C.UTF-8
export COLORTERM=truecolor
export XKB_DEFAULT_MODEL=pc105
export XKB_DEFAULT_LAYOUT=us
stty iutf8
```
None of this helped
